rust/repositories.bzl

```diff
@@ -493,6 +511,32 @@ def _load_rust_compiler(ctx):
 
     return compiler_build_file
 
+def _load_rust_src(ctx):
+    """Loads the rust source code. Used by the rust-analyzer rust-project.json generator.
+
+    Args:
+        ctx (ctx): A repository_ctx.
+    Returns:
+        string: The BUILD file contents for the rust source code
+    """
+    tool_suburl = produce_tool_suburl("rustc", "src", ctx.attr.version, ctx.attr.iso_date)
+    static_rust = ctx.os.environ.get("STATIC_RUST_URL", "https://static.rust-lang.org")
+    url = "{}/dist/{}.tar.gz".format(static_rust, tool_suburl)
+
+    tool_path = produce_tool_path("rustc", "src", ctx.attr.version)
+    archive_path = tool_path + ".tar.gz"
+    ctx.download(
+        url,
+        output = archive_path,
+        sha256 = ctx.attr.sha256s.get(tool_suburl) or FILE_KEY_TO_SHA.get(tool_suburl) or "",
```

Comment:
Without the empty string you are forced to supply a sha256 in your WORKSPACE file or there is an error. Falling back to the empty string allows you to avoid setting a sha256 (and bazel will tell you the sha it used in an info), and matches the behavior of all the other tools we fetch.